src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -707,6 +724,29 @@ export class CellModel implements ICellModel {
 		return source;
 	}
 
+	// Run ADS commands via a notebook. Configured by a setting (turned off by default).
+	private checkForAdsCommandMagic(): boolean {
+		if (!this._isCommandExecutionSettingEnabled) {
+			return false;
+		}
+
+		let executeCommandMagic = '%%AZDATA_EXECUTE_COMMAND';
+		if (this._source && this._source.length) {
+			if (this._source[0].startsWith(executeCommandMagic)) {
```

Comment:
You're only checking the first line? Looking at documentation it seems like it's valid to have magics on any line and I could see a case where someone wants to run a command only if a certain case has occurred or something.

Reply:
Line magics start with % and can be called on any line; cell magics start with %%. I'm not aware of any cell magics that are defined on any lines other than the first 😄

Reply:
Ah cool, one of these days I'll go through the Jupyter documentation and get to know all this stuff!

Why did you go with a cell magic? This seems to be more of a line magic type thing (what if I want to execute a command with the result of some query or something? I guess you could store them in variables and re-use them but that seems a bit convoluted)